kotlin/reflect/src/main/java/com/squareup/moshi/kotlin/reflect/KotlinJsonAdapter.kt

```diff
@@ -133,46 +146,59 @@ internal class KotlinJsonAdapter<T>(
     writer.endObject()
   }
 
-  override fun toString() = "KotlinJsonAdapter(${constructor.returnType})"
+  override fun toString() = "KotlinJsonAdapter(${constructor.type.canonicalName})"
 
   data class Binding<K, P>(
-    val name: String,
-    val jsonName: String?,
-    val adapter: JsonAdapter<P>,
-    val property: KProperty1<K, P>,
-    val parameter: KParameter?,
-    val propertyIndex: Int
+      val name: String,
+      val jsonName: String?,
+      val adapter: JsonAdapter<P>,
+      val property: PropertyData,
+      val propertyIndex: Int = property.parameter?.index ?: -1
   ) {
-    fun get(value: K) = property.get(value)
+
+    fun get(value: K): Any? {
+      property.jvmGetter?.let { getter ->
+        return getter.invoke(value)
+      }
+      property.jvmField?.let {
+        return it.get(value)
+      }
+
+      return null
```

Comment:
is this reachable?

Reply:
I suppose not, will make it an error call instead 20f762f